architecture/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,457 @@
+        namespace: 'Mycompany.Messaging'
+        apiVersion: '2024-09-01-preview'
+        resourceTypes:
+            'Mycompany.Messaging/plaidResource':
+                name: 'Name of the resource'
+                    description: 'Plaid Messaging service' 
+                    properties: 
+                    required: ['host', 'port']
+                    host:
+                        type: 'string'
+                        description: 'Hostname'
+                    
+                    port:
+                        type: 'string'
+                        description: 'Port''
+            ...
+        ```
+
+    1. Deb registers the schema in Radius
+        
+        ```bash
+        rad resource-provider register Mycompany.Messaging 
+        ```
+        Radius complies the schema and registers all the custom resource type including Plaid in UCP
+
+    1. Deb lists the resource types he has in his environment
+
+        ```bash
+        rad resource-provider list 
+        ```
+        Radius lists the resource providers that Deb has registered in his environment
+
+     1. Deb publishes the bicep types to a registry
+
+        ```bash
+        rad publish-bicep-types Mycompany.Messaging --registry myregistry
+        ```
+
+        The `plaid.yaml` is complied into open API specs; the `index.json` and `types.json` files are generated and published to the registry for the custom resource type Plaid and the new resource provider is added to `bicepconfig.json`
+
+        ```json
+        {
+            "experimentalFeaturesEnabled": {
+                "extensibility": true,
+                "extensionRegistry": true,
+                "dynamicTypeLoading": true
+            },
+            "extensions": {
+                "radius": "br:biceptypes.azurecr.io/radius:latest",
+                "aws": "br:biceptypes.azurecr.io/aws:latest"
+                "Mycompany": "br:biceptypes.azurecr.io/mycompany:latest"
+            }
+        }
+        ```
+    1. Create a recipe for the custom resource type Plaid
+
+        ```bash
+        rad recipe register plaid-dev –resource-type “Mycompany.Messaging/Plaid" --template-kind bicep --template-path "<myregistry>/plaid:latest"
+        ```
+    1. Create a app.bicep with plaid resource type
+        ```bicep
+        param plaidName string
+        param plaidDescription string
+        param plaidQueue string
+        resource plaid 'Mycompany.Messaging/Plaid@2024-08-09-preview' = {
+            name: plaidName
+            properties: {
+            recipe:{
+                name: 'plaid-dev'
+            }
+            }
+        }
+        ```
+    1. Deb uses the rad CLI to deploy the application with the custom resource type Plaid
+        
+        ```bash
+        rad deploy app.bicep 
+        ```
+    The application is deployed with the custom resource type Plaid in Radius. Deb can iterate on the changes, compile and test the application with the custom resource type.
+
+    
```

Comment:
**Note - This is an illustrative example of how the experience would be with typespec which may be implemented in future based on feedback**